# If we're building with clang then we want to ensure that PIC is being set, not PIE

diff -up qtbase-everywhere-src-5.15.1/src/corelib/global/qglobal.h.3~ qtbase-everywhere-src-5.15.1/src/corelib/global/qglobal.h
--- qtbase-everywhere-src-5.15.1/src/corelib/global/qglobal.h.3~	2020-09-09 18:04:25.250844842 +0200
+++ qtbase-everywhere-src-5.15.1/src/corelib/global/qglobal.h	2020-09-09 18:05:07.260388851 +0200
@@ -1282,7 +1282,7 @@ Q_CORE_EXPORT QT_DEPRECATED_VERSION_X_5_
 #define QT_MODULE(x)
 
 #if !defined(QT_BOOTSTRAPPED) && defined(QT_REDUCE_RELOCATIONS) && defined(__ELF__) && \
-    (!defined(__PIC__) || (defined(__PIE__) && defined(Q_CC_GNU) && Q_CC_GNU >= 500))
+    (!defined(__PIC__) || (defined(__PIE__) && defined(Q_CC_GNU) && Q_CC_GNU >= 500) || (defined(__PIE__) && defined(__clang__)))
 #  error "You must build your code with position independent code if Qt was built with -reduce-relocations. "\
          "Compile your code with -fPIC (and not with -fPIE)."
 #endif
